File: 71.js
var simplifyPath = function (path) {
    let simplePath = []
    let str = ""

    for (i = 0; i < path.length; i++) {
        if (path[i] === "/") {
            if (str == "..") {
                simplePath.pop()
                str = ""
            } else if (str == ".") {
                str = ""
                continue;
            } else {
                if (str != "") simplePath.push(str);
                str = ""
            }
        } else {
            str += path[i];
        }
    }

    if (str == "..") {
        simplePath.pop()
    } else {
        if (str != "" && str != ".") simplePath.push(str);
    }
    return ("/" + simplePath.join("/"))

};

File: 71.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        stack=[]
        p = path.split("/")
        hashMap={
            "":1,
            ".":1,
            "..":1,
            "/":1
        }

        for x in p:
            if x not in hashMap:
                stack.append(x)
            elif x == ".." and len(stack)>0:
                stack.pop()
        if len(stack)==0:
            return "/"
        return "/" + "/".join(stack)
        